engines/net.c

```diff
@@ -1110,12 +1155,50 @@ static int fio_netio_setup_connect_unix(struct thread_data *td,
 	return 0;
 }
 
+static int fio_netio_setup_connect_vsock(struct thread_data *td,
+					const char *host, unsigned short port)
+{
+#ifdef CONFIG_VSOCK
+	struct netio_data *nd = td->io_ops_data;
+	struct sockaddr_vm *addr = &nd->addr_vm;
+	int cid;
+
+	if (!host) {
+		log_err("fio: connect with no host to connect to.\n");
+		if (td_read(td))
+			log_err("fio: did you forget to set 'listen'?\n");
+
+		td_verror(td, EINVAL, "no hostname= set");
+		return 1;
+	}
+
+	addr->svm_family = AF_VSOCK;
+	addr->svm_port = port;
+
+	if (host) {
+		cid = atoi(host);
+		if (cid < 0 || cid > UINT32_MAX) {
+			log_err("fio: invalid CID %d\n", cid);
+			return 1;
+		}
+		addr->svm_cid = cid;
+	}
+
+	return 0;
+#else
+	td_verror(td, -EINVAL, "vsock not supported");
+	return 1;
+#endif
+}
+
 static int fio_netio_setup_connect(struct thread_data *td)
 {
 	struct netio_options *o = td->eo;
 
 	if (is_udp(o) || is_tcp(o))
 		return fio_netio_setup_connect_inet(td, td->o.filename,o->port);
+	else if (is_vsock(o))
+	    return fio_netio_setup_connect_vsock(td, td->o.filename, o->port);
```

Comment:
You still have whitespace issues here, please only use tabs, never spaces. I suspect you have your editor set for tab == 4 spaces.